shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -340,6 +392,77 @@ - (BOOL)handleKeyEvent:(NSEvent*)event {
   return [_textInputContext handleEvent:event];
 }
 
+#pragma mark -
+#pragma mark NSResponder
+
+- (void)keyDown:(NSEvent*)event {
+  [self.flutterViewController keyDown:event];
+}
+
+- (void)keyUp:(NSEvent*)event {
+  [self.flutterViewController keyUp:event];
+}
+
+- (BOOL)performKeyEquivalent:(NSEvent*)event {
+  return [self.flutterViewController performKeyEquivalent:event];
+}
+
+- (void)flagsChanged:(NSEvent*)event {
+  [self.flutterViewController flagsChanged:event];
+}
+
+- (void)mouseEntered:(NSEvent*)event {
```

Comment:
Why are you forwarding `mouseEntered` and `mouseExited`? That seems very likely to break hover state tracking; mousing out of a text field isn't the same thing as mousing out of the Flutter view.